### PR TITLE
Refactor the remaining uses of indexSettings(Version) to IndexVersion

### PR DESCRIPTION
**Pull request by thecoop**
Refactor the remaining tests that use `indexSettings(Version)`.

It looks like the two security tests don't actually do anything anymore - the join validator they're testing doesn't care about nodes at all, it's just checking FIPS. So I've removed them completely.

### CODE REVIEW

**Comment by ldematte** [comment]
Let me ask a clarification: the older version of the tests used always the (global) oldVersion, which was random between the version with minimum index compatibility and current; the new code does the same, right? So both tests (new and old) have some randomness (the version) in them, correct?

**Comment by ldematte** [comment]
> This _shouldn't_ matter, as they've been completely separated anyway...
So it is good to have them randomly independent now: if tests break, it is unexpected

**Review by ldematte — APPROVED** [review]
LGTM

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java [hunk 1]
```diff
@@ -472,39 +472,6 @@ public void testIndexJoinValidator_FullyCurrentCluster() throws Exception {
         joinValidator.accept(node, clusterState);
     }
 
-    public void testIndexUpgradeValidatorWithUpToDateIndex() throws Exception {
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java:475]: Shouldn't the above `testIndexJoinVAlidator_FullyCurrentCluster()` be removed as well?

**Review by albertzaharovits — APPROVED** [review]

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java (hunk 1 above)
Comment (thecoop) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java:475]: Good spot, removed

**Pull request merged**